openff/toolkit/tests/test_molecule.py

```diff
@@ -3418,6 +3420,25 @@ def test_assign_partial_charges(self, toolkit, method):
         recomputed_charges = molecule._partial_charges
         np.testing.assert_allclose(initial_charges, recomputed_charges, atol=0.002)
 
+    def test_partial_charges_setter_type_conversion(self):
+        molecule = Molecule.from_smiles("C")
+        int_charges = np.zeros(molecule.n_atoms, dtype=int)
+        molecule.partial_charges = int_charges * unit.elementary_charge
+        assert molecule.partial_charges.dtype == float
+
+    @pytest.mark.parametrize(
+        "value, error",
+        [
+            (3, IncompatibleTypeError),
+            (np.zeros(5), IncompatibleTypeError),
+            (np.zeros(2), IncompatibleShapeError),
```

Comment:
```suggestion
            (np.zeros(2), IncompatibleShapeError),
            (np.zeros(2) * unit.elementary_charge, IncompatibleShapeError),

```
Unless it's already covered (I didn't look too hard) it'd be nice to ensure the shape failure is caught whether or not it's the correct data type